web/src/engine/predictive-text/templates/src/trie-decoder.ts

```diff
@@ -0,0 +1,35 @@
+import { type Wordform2Key } from './common.js';
+import { decompressNode } from './trie-compression.js';
+import { type InternalNode } from './trie.js';
+
+export function buildDecoder(encodedRoot: string, searchTermKeyer: Wordform2Key) {
+  const root = decompressNode(encodedRoot, searchTermKeyer, 0) as InternalNode;
+  retrofitInternalNode(root, searchTermKeyer);
+
+  return root;
+}
+
+function retrofitInternalNode(node: InternalNode, searchTermKeyer: Wordform2Key) {
+  const originalChildren = node.children;
+
+  node.children = {};
+  const children = node.children;
+  for(let char of node.values) {
+    const child = originalChildren[char] as string;
+    Object.defineProperty(children, char, {
```

Comment:
Consider this a lazy-decompression strategy.

Once we _actually_ need to reference a value, _then_ we decode it.